### PR TITLE
Add ansible-lint to GitHub Actions and associated changes

### PR DESCRIPTION
**Pull request by jamesholland-uk**
## Description
Adds `ansible-lint` to CI, along with a relevant config file.
In order for `ansible-lint` to pass successfully on the Collection, this PR also includes:
- Updates to versions of Ansible tested with `sanity` tests in CI, to cater for new releases and EOL'd releases
- Updates to versions of Python used for `sanity` with a new matrix
- Updates to documentation to match arguments specs
- Updates to match deprecation items into `runtime.yml`
- Change to the input choices for `forward_next_hop_type` of `panos_pbf_rule`

## Motivation and Context
`ansible-lint` is required per requirements for Ansible Collection certification since Jun 2023

## How Has This Been Tested?
CI tested in fork of the repo.
Change to the input choices for `forward_next_hop_type` of `panos_pbf_rule` tested locally.

## Screenshots (if appropriate)
Screenshot 2023-08-21 at 16 29 10

## Types of changes
- Bug fix (non-breaking change which fixes an issue)

## Checklist
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

File: plugins/modules/panos_pbf_rule.py [hunk 1]
```diff
@@ -246,7 +246,7 @@ def main():
             ),
             forward_vsys=dict(),
             forward_egress_interface=dict(),
-            forward_next_hop_type=dict(choices=[None, "ip-address", "fqdn"]),
+            forward_next_hop_type=dict(choices=["None", "ip-address", "fqdn"]),
```
Comment (shinmog) [inline plugins/modules/panos_pbf_rule.py:249]: Python `None` is allowed here.  Doing a string of `"None"` is not right.

File: plugins/modules/panos_sag.py [hunk 2]
```diff
@@ -84,6 +84,7 @@
         type: list
         elements: str
         required: false
+        default: []
```
Comment (shinmog) [inline plugins/modules/panos_sag.py:87]: I'm not entirely sure this is correct....?  Did you verify that a list defaults to an empty list if it's not specified?  I would have expected Python `None` here.

File: tests/config.yml [hunk 3]
```diff
@@ -37,4 +37,3 @@ modules:
   # These settings only apply to modules/module_utils.
   # It is not possible to declare supported Python versions for controller-only code.
   # All Python versions supported by the controller must be supported by controller-only code.
-
```
Comment (shinmog) [inline tests/config.yml:40]: Seems an unintentional removal?

**Review by shinmog — COMMENTED** [review]
Just a few things, more questions than anything.

I got excited seeing the removal of "ignore-2.9.txt" but then that got squashed when I saw the addition of "ignore-2.14.txt" and "ignore-2.15.txt", lol.

File: tests/config.yml (hunk 3 above)
Comment (jamesholland-uk) [reply tests/config.yml:40]: That was to pass the new `ansible-lint` tests:

```
WARNING  Listing 112 violation(s) that are fatal
.
.
.
yaml[new-line-at-end-of-file]: No new line character at the end of file
tests/config.yml:40
.
.
.
                       Rule Violation Summary                        
 count tag                           profile    rule associated tags 
     3 yaml[comments]                basic      formatting, yaml     
     2 yaml[indentation]             basic      formatting, yaml     
     2 yaml[new-line-at-end-of-file] basic      formatting, yaml     
     3 yaml[trailing-spaces]         basic      formatting, yaml     
     1 yaml[truthy]                  basic      formatting, yaml     
    99 sanity[cannot-ignore]         production idiom                
     2 fqcn[deep]                    production formatting (warning) 
```

**Review by shinmog — APPROVED** [review]
lgtm

**Pull request merged**